base/sparse/sparsematrix.jl

```diff
@@ -1403,13 +1403,19 @@ sparse(S::UniformScaling, m::Integer, n::Integer=m) = speye_scaled(S.λ, m, n)
 # map/map! entry points
 function map!{Tf,N}(f::Tf, C::SparseMatrixCSC, A::SparseMatrixCSC, Bs::Vararg{SparseMatrixCSC,N})
     _checksameshape(C, A, Bs...)
+    return map_nocheck!(f, C, A, Bs...)
```

Comment:
Maybe this could be tied to the bounds checking mechanism? Or would it be an abuse?